src/main/java/htsjdk/samtools/util/OverlapDetector.java

```diff
@@ -80,25 +81,35 @@ public void addLhs(final T object, final Locatable interval) {
         }
 
         final int start = interval.getStart() + this.lhsBuffer;
-        final int end   = interval.getEnd()   - this.lhsBuffer;
+        final int end = interval.getEnd() - this.lhsBuffer;
 
-        final Set<T> newValue = Collections.singleton(object);
         if (start <= end) {  // Don't put in sequences that have no overlappable bases
-            final Set<T> alreadyThere = tree.put(start, end, newValue);
-            if (alreadyThere != null) {
-                if( alreadyThere.size() == 1){
-                    Set<T> mutableSet = new HashSet<>(2);
-                    mutableSet.addAll(alreadyThere);
-                    mutableSet.add(object);
-                    tree.put(start, end, mutableSet);
-                } else {
-                    alreadyThere.add(object);
-                    tree.put(start, end, alreadyThere);
-                }
-            }
+            tree.merge(start, end,
+                                  Collections.singleton(object),
+                                  mergeSetsAccountingForSingletons());
         }
     }
 
+    /**
+     * merge two Sets, assumes sets of size 1 are immutale
+     */
+    private static <T> BiFunction<Set<T>, Set<T>, Set<T>> mergeSetsAccountingForSingletons() {
+        return (newValue, oldValue) -> {
+            // Sets of size 1 are immutable SingletonSets so we have to make a new
+            // mutable one to add to
+            if (oldValue.size() == 1) {
+                Set<T> newMutableSet = new HashSet<>();
```

Comment:
can replace entire if statement with:
```
final Set<T> newMutableSet = oldValue.size() == 1 ? new HashSet<>() : oldValue;
newMutableSet.addAll(oldValue);
newMutableSet.addAll(newValue);
return newMutableSet;
```

Reply:
Oh right, clever, it's a set so you can unconditionally add the same thing over again...